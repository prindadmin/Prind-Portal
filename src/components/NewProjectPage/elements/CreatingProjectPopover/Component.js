import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
