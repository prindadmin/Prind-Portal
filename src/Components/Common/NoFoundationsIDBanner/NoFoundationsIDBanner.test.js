import React from 'react';
import { shallow } from 'enzyme';

import Component from './NoFoundationsIDBanner'

it('Should not render if user has Foundations ID', () => {
  const props = {
    user: {
      details: {
        foundationsID: '123'
      }
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should render if user has Foundations ID', () => {
  const props = {
    user: {
      details: {}
    }
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Should not render if not enough user details provided', () => {
  const props = {
    user: {}
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
