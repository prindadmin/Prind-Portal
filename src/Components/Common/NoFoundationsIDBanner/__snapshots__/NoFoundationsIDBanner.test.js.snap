// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render if not enough user details provided 1`] = `""`;

exports[`Should not render if user has Foundations ID 1`] = `""`;

exports[`Should render if user has no Foundations ID 1`] = `
<div
  className="row full-width"
  id="no-foundations-id-banner"
  onClick={[Function]}
>
  <p>
    You do not have a Foundations ID and therefore cannot sign any documents.  Click here to find out more...
  </p>
</div>
`;
